### ISSUE TITLE
Move components and hooks to dedicated library

### ISSUE THREAD

**Issue by Awem** [twilio/twilio-video-app-react#272]
**What is the feature that you would like to see in the app? Please describe.**
This repo provides a lot of useful hooks and components that encapsulate Twilio-specific requirements. If these hooks and components could be provided as a separate SDK, they could greatly facilitate developing apps that include a Twilio video chat.

I thinks using these hooks and components only for demonstration purposes is missing their potential.

**Is your feature request related to a problem? Please describe.**
I am also developing a Twilio video app, that currently uses somewhat similar hooks. I could definitely make great use of those hooks.

**Describe alternatives you've considered**
I could of course copy some of the code to my own repository. Some of the hooks I could use directly. Some I would alter. This would work, but would be tedious and would not stay in sync with upstream changes.

**Additional context**
Let me know what you think. Maybe I can help with the refactoring.

**Comment by timmydoza** [comment]
Thanks for the feature request @Awem! I've let my team know that there is community interest in this feature. At the moment, this feature is not on our roadmap, but I'll be sure to update this ticket if that changes. Thanks!

**Comment by yareyaredesuyo** [comment]
Yeah, changing to monorepo and offering some useful features as a library maybe helpful.

like this project

https://github.com/backstage/backstage

**Comment by bhatvikrant** [comment]
This would be so helpful if implemented!

**Comment by tlmdouellet** [comment]
We would definively use it. Currently, we fork the whole repo in our project and change some lines to make it re-usable in our app like a regular library. This is just really hard to maintain up to date
